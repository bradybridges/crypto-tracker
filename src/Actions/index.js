export const updateCryptos = (cryptos) => ({
  type: 'UPDATE_CRYPTOS',
  cryptos,
});

export const updateError = (error) => ({
  type: 'UPDATE_ERROR',
  error,
});

export const updateTimeFrame = (timeFrame) => ({
  type: 'UPDATE_TIME_FRAME',
  timeFrame,
});

export const updateTrackedCoins = (symbols) => ({
  type: 'UPDATE_TRACKED_COINS',
  symbols,
});

export const updatePortfolio = (portfolio) => ({
  type: 'UPDATE_PORTFOLIO',
  portfolio,
});
