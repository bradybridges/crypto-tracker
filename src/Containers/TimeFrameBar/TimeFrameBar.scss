#time-frame-table {
  width: 50%;
  justify-self: center;
  border-collapse: collapse;
  margin: 0 auto;
  background-color: #3e3e3ed4;
  text-align: center;
  margin-bottom: 20px;
}

.active {
  background-color: blue;
}