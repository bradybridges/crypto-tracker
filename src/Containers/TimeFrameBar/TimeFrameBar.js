import React, { Component } from 'react';
import './TimeFrameBar.scss';
import { updateTimeFrame } from '../../Actions/index';
import { connect } from 'react-redux';

export class TimeFrameBar extends Component {

  handleClick = e => {
    e.preventDefault();
    this.props.updateTimeFrame(e.target.id);
  }

  render() {
    const oneHourClass = this.props.timeFrame === '1h' ? 'active' : 'inactive';
    const oneDayClass = this.props.timeFrame === '1d' ? 'active' : 'inactive';
    const oneMonthClass = this.props.timeFrame === '30d' ? 'active' : 'inactive';
    const oneYearClass = this.props.timeFrame === '365d' ? 'active' : 'inactive';
    return (
      <table id='time-frame-table'>
        <tbody>
          <tr>
            <td className={oneHourClass} onClick={this.handleClick} id='1h'>1 HR</td>
            <td className={oneDayClass} onClick={this.handleClick} id='1d'>1 D</td>
            <td className={oneMonthClass} onClick={this.handleClick} id='30d'>30 D</td>
            <td className={oneYearClass} onClick={this.handleClick} id='365d'>1 YR</td>
          </tr>
        </tbody>
      </table>
    );
  };
};

export const mapDispatchToProps = dispatch => ({
  updateTimeFrame: timeFrame => dispatch( updateTimeFrame(timeFrame)),
});

export const mapStateToProps = state => ({
  timeFrame: state.timeFrame,
});

export default connect(mapStateToProps, mapDispatchToProps)(TimeFrameBar);