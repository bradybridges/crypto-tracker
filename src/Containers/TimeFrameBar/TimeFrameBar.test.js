import React from 'react';
import { shallow } from 'enzyme';
import { updateTimeFrame } from '../../Actions/index';
import { TimeFrameBar, mapStateToProps, mapDispatchToProps } from './TimeFrameBar'; 

describe('TimeFrameBar', () => {
  let wrapper;
  const mockUpdateTimeFrame = jest.fn();
  beforeEach(() => {
    wrapper = shallow(<TimeFrameBar updateTimeFrame={mockUpdateTimeFrame} />);
  })
  it('should match the snapshot', () =>{
    expect(wrapper).toMatchSnapshot();
  });

  it('handle click should call updateTimeFrame', () => {
    const mockEvent = {
      preventDefault: jest.fn(),
      target: {id: '365d'},
    }
    wrapper.instance().handleClick(mockEvent);
    expect(wrapper.instance().props.updateTimeFrame).toHaveBeenCalledWith('365d');
  });

  it('when a time frame button is clicked it should run handleClick', () => {
    const mockEvent = {
      preventDefault: jest.fn(),
      target: {id: '1d'}
    }
    wrapper.instance().props.updateTimeFrame.mockClear();
    wrapper.find('td').at(1).simulate('click', mockEvent);
    expect(wrapper.instance().props.updateTimeFrame).toHaveBeenCalledWith('1d');
  });
});

describe('mapDispatchToProps', () => {
  it('should return updateTimeFrame and dispatch the method properly', () => {
    const mockDispatch = jest.fn();
    const actionToDispatch = updateTimeFrame('365d');
    const mappedDispatch = mapDispatchToProps(mockDispatch);
    mappedDispatch.updateTimeFrame('365d');
    expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
  });
});

describe('mapStateToProps', () => {
  it('should map timeFrame to props', () => {
    const mockState = {
      timeFrame: '365d',
      error: 'An error occured',
    }
    const expectedState = {timeFrame: '365d'};
    const mappedState = mapStateToProps(mockState);
    expect(mappedState).toEqual(expectedState);
  });
});