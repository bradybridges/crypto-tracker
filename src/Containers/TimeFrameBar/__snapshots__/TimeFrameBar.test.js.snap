// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TimeFrameBar should match the snapshot 1`] = `
<table
  id="time-frame-table"
>
  <tbody>
    <tr>
      <td
        className=""
        id="1h"
        onClick={[Function]}
      >
        1HR
      </td>
      <td
        className=""
        id="1d"
        onClick={[Function]}
      >
        1D
      </td>
      <td
        className=""
        id="30d"
        onClick={[Function]}
      >
        30D
      </td>
      <td
        className=""
        id="365d"
        onClick={[Function]}
      >
        1YR
      </td>
    </tr>
  </tbody>
</table>
`;
